src/input/SimModSuite.h

```diff
@@ -416,43 +418,39 @@ void setCases(pGModel model, pACase &meshCase, pACase &analysisCase, MeshAttribu
 
     //With this code we can tag any BC, not only 1,3,5
     int numFaces = GM_numFaces(model);
-    std::set<int> UniquefaceBound(AnalysisAtt.faceBound, AnalysisAtt.faceBound + numFaces);
+    std::set<int> UniquefaceBound;
+    for (auto const& f : AnalysisAtt.faceBound) {
+      UniquefaceBound.insert(f.second);
+    }
     int numBC = UniquefaceBound.size();
-    pAttInfoVoid iBC[numBC];
-    pModelAssoc aBC[numBC];
-    std::set<int>::iterator it = UniquefaceBound.begin();
-
-    std::map<int,int> LUT;
+    std::map<int, pModelAssoc> aBC;
 
-    for (int i = 0; i < numBC; i++) {
+    for (auto const& ufb : UniquefaceBound) {
        std::string strboundaryCondition="BC";
-       char buff[100];
-       snprintf(buff, sizeof(buff), "%09d", (*it));
-       std::string sNumber = buff;
-       //std::string sNumber = std::to_string((*it));
-       LUT[(*it)]=i;
-       it++;
-       iBC[i] = AMAN_newAttInfoVoid(attMngr,"BC","boundaryCondition");
+       std::ostringstream stringstream;
+       stringstream << std::setw(9) << std::setfill('0') << ufb;
+       std::string sNumber = stringstream.str();
+
+       pAttInfoVoid iBC = AMAN_newAttInfoVoid(attMngr,"BC","boundaryCondition");
        strboundaryCondition.append(sNumber);
-       AttNode_setImageClass((pANode)iBC[i],strboundaryCondition.c_str());
-       AttCase_addNode(analysisCase,(pANode)iBC[i]);
-       aBC[i] = AttCase_newModelAssoc(analysisCase,(pANode)iBC[i]);
+       AttNode_setImageClass((pANode)iBC,strboundaryCondition.c_str());
+       AttCase_addNode(analysisCase,(pANode)iBC);
+       aBC[ufb] = AttCase_newModelAssoc(analysisCase,(pANode)iBC);
     }
 
     pGEntity face;
```

Comment:
Here face is uninitialized - why isn't this done in the loop? In this case, it could be const as well

Reply:
Legacy :man_shrugging:

Reply:
But I've fixed it.

Reply:
I thought that initializing over and over would be slower than initializing once.